### PR TITLE
chore(workflows):[TRI-1069] blacklisted and separated workflows

### PR DESCRIPTION
**Pull request by ds-mmaul**

### CODE REVIEW

File: .github/workflows/irs-build.yml [hunk 1]
```diff
@@ -34,20 +45,29 @@ jobs:
           key: ${{ runner.os }}-m2-${{ hashFiles('**/pom.xml') }}
           restore-keys: ${{ runner.os }}-m2
```
Comment (ds-jhartmann) [inline .github/workflows/irs-build.yml:47]: Please keep the normal maven build `mvn clean verify --batch-mode`

File: .github/workflows/irs-build.yml [hunk 2]
```diff
@@ -34,20 +45,29 @@ jobs:
...
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0  # Shallow clones should be disabled for a better relevancy of sonar analysis
+
       - name: Cache SonarCloud packages
         uses: actions/cache@v3
         with:
           path: ~/.sonar/cache
           key: ${{ runner.os }}-sonar
           restore-keys: ${{ runner.os }}-sonar
```
Comment (ds-jhartmann) [inline .github/workflows/irs-build.yml:63]: add the 'Cache maven packages' step here as well

**Review by ds-jhartmann — CHANGES_REQUESTED** [review]

File: .github/workflows/irs-build.yml (hunk 1 above)
Comment (ds-mmaul) [reply .github/workflows/irs-build.yml:47]: readded the maven build step with run command `mvn clean verify --batch-mode`

File: .github/workflows/irs-build.yml (hunk 2 above)
Comment (ds-mmaul) [reply .github/workflows/irs-build.yml:73]: added 'Cache maven packages' to the 'analyze with sonar'- workflow

File: .github/workflows/irs-build.yml [hunk 3]
```diff
@@ -34,20 +45,39 @@ jobs:
           key: ${{ runner.os }}-m2-${{ hashFiles('**/pom.xml') }}
           restore-keys: ${{ runner.os }}-m2
 
+      - name: Build with Maven
+        run: |
+          mvn clean verify --batch-mode
+
+  analyze_with_Sonar:
+    runs-on: ubuntu-latest
+    needs: build
```
Comment (ds-jhartmann) [inline .github/workflows/irs-build.yml:54]: You can remove the 'needs: build' requirement. Sonar scan is not dependent on the build job and they can run in parallel.

**Review by ds-jhartmann — CHANGES_REQUESTED** [review]
One last thing

File: .github/workflows/irs-build.yml [hunk 4]
```diff
@@ -34,20 +45,38 @@ jobs:
           key: ${{ runner.os }}-m2-${{ hashFiles('**/pom.xml') }}
           restore-keys: ${{ runner.os }}-m2
 
+      - name: Build with Maven
+        run: |
+          mvn clean verify --batch-mode
+
+  analyze_with_Sonar:
+    runs-on: ubuntu-latest
+    steps:
```
Comment (ds-jhartmann) [inline .github/workflows/irs-build.yml:54]: Please also add jdk 17

**Review by ds-jhartmann — CHANGES_REQUESTED** [review]

File: .github/workflows/irs-build.yml (hunk 4 above)
Comment (ds-mmaul) [reply .github/workflows/irs-build.yml:53]: added jdk 17

**Review by ds-jhartmann — APPROVED** [review]

**Pull request merged**